python/lsst/ip/diffim/getTemplate.py

```diff
@@ -705,3 +719,157 @@ def run(self, coaddExposures, bbox, wcs):
+                                                                     inputs['wcs'])
+        self.checkPatchList(patchList)
+        outputs = self.run(coaddExposuresRefs, inputs['bbox'], inputs['wcs'], inputs['visitInfo'])
+        butlerQC.put(outputs, outputRefs)
+
+    def checkPatchList(self, patchList):
+        """Check that all of the DcrModel subfilters are present for each patch.
+
+        Parameters
+        ----------
+        patchList : `dict`
+            Dict of the patches containing valid data for each tract
+
+        Raises
+        ------
+        RuntimeError
+            If the number of exposures found for a patch does not match the number of subfilters.
+        """
+        for tract in patchList:
+            for patch in set(patchList[tract]):
+                if patchList[tract].count(patch) != self.config.numSubfilters:
+                    raise RuntimeError("Invalid number of DcrModel subfilters found: %d vs %d expected",
+                                       patchList[tract].count(patch), self.config.numSubfilters)
+
+    def getCoaddPatch(self, coaddExposureRef, visitInfo):
+        """Retrieve the DcrModel for a patch in a single subfilter.
+
+        Parameters
+        ----------
+        coaddExposureRef : DeferredDatasetHandle to `lsst.afw.image.Exposure`
+            A data reference to a single coadd to be mosaiced.
+        visitInfo : `lsst.afw.image.VisitInfo`
+            Metadata for the exposure.
+
+        Returns
+        -------
+        coaddPatch : `lsst.afw.image.Exposure`
+            A single coadd exposure.
+        """
+        dcrModel = DcrModel.fromQuantum(coaddExposureRef,
+                                        self.config.effectiveWavelength,
+                                        self.config.bandwidth)
+        # The edge pixels of the DcrCoadd may contain artifacts due to missing data.
+        # Each patch has significant overlap, and the contaminated edge pixels in
+        # a new patch will overwrite good pixels in the overlap region from
+        # previous patches.
+        # Shrink the BBox to remove the contaminated pixels,
+        # but make sure it is only the overlap region that is reduced.
+        dcrBBox = dcrModel.bbox
+        dcrBBox.grow(-self.config.templateBorderSize)
+        singleSubfilter = dcrModel.buildMatchedExposure(bbox=dcrBBox,
+                                                        visitInfo=visitInfo)
+        # Each subfilter is loaded separately, and will be averaged together in
+        # the ``lsst.afw.math.statisticsStack`` call. Since the subfilter images
+        # will be averaged instead of added together we need to multiply each
+        # image by the number of subfilters.
+        singleSubfilter *= self.config.numSubfilters
+        return singleSubfilter
+
+
+class GetMultiTractCoaddTemplateConfig(GetTemplateConfig):
+    pass
+
+
+class GetMultiTractCoaddTemplateTask(GetTemplateTask):
```

Comment:
Would you also do a search the stack for `GetMultiTractCoaddTemplateTask` and replace our invocations of it too? I know you'll find it in `ap_pipe` and `drp_pipe`.  Without that, this will just add warnings to our logs.